index.rst

```diff
@@ -632,15 +641,76 @@ The disassembler function signature is:
             The object that will be disassembled into component parts.
         dataId : dict
             The dataId that is being used to persist this object.
-        componentDict : dict
-            A dict to populate with the components that should be persisted for
-            this composite.
+        componentInfo : dict of componentDatasetName to ComponentInfo
+            A dict of ComponentInfo instances to populate with the components
+            that should be persisted for this composite. Keys are the component
+            names as defined by the datasetType in the policy. Values are
+            ComponentInfo objects.
 
         Returns
         -------
         None
         """
 
+ComponentInfo
+"""""""""""""
+
+ComponentInfo is a class used when assembling and disassembling a composite
+object in butler. It contains information about a component of a composite
+object and has a slot for passing the component object into the assembler and for
+the disssembler to pass the component object out.
+
+ComponentInfo is used as an input to assemblers and disassemblers (which are
+part of the butler public API).
+
+Some ComponentInfo parameters are populated with information from the policy and
```

Comment:
Do you anticipate that the setter and getter may be useful for more than the generic assembler/disassembler?  If not, it might be nice to pass them separately to that function (i.e. the generic assembler/disassembler would have a different signature than a policy-specified one) which would allow simplification of the `ComponentInfo` class (perhaps even back to eliminating it altogether).